src/hotspot/share/gc/shared/collectedHeap.cpp

```diff
@@ -633,35 +607,52 @@ void CollectedHeap::post_initialize() {
+  LogTarget(Info, cpu) cpuLog;
+  if (!os::is_thread_cpu_time_supported() || !cpuLog.is_enabled()) {
+    return;
+  }
+
+  double process_cpu_time = os::elapsed_process_cpu_time();
+  if (process_cpu_time == 0 || process_cpu_time == -1) {
+    // 0 can happen e.g. for short running processes with
+    // low CPU utilization
+    return;
+  }
 
-    double usage;
-    if (gc_cpu_time > process_cpu_time ||
-        process_cpu_time == 0 || gc_cpu_time == 0) {
-      // This can happen e.g. for short running processes with
-      // low CPU utilization
-      usage = 0;
-    } else {
-      usage = 100 * gc_cpu_time / process_cpu_time;
+  double vm_thread_cpu_time = (double) CPUTimeUsage::Runtime::vm_thread() / NANOSECS_PER_SEC;
+  CPUTimeUsage::GCStatistics gc_stats = CPUTimeUsage::GC::statisics();
+  double gc_cpu_time = (double) gc_stats.total / NANOSECS_PER_SEC;
+  double gc_threads_cpu_time = (double) gc_stats.gc_threads / NANOSECS_PER_SEC;
+  double gc_vm_thread_cpu_time = (double) gc_stats.vm_thread / NANOSECS_PER_SEC;
+
+  if (gc_cpu_time < process_cpu_time) {
+    cpuLog.print("=== CPU time Statistics =============================================================");
+    cpuLog.print("                                                                            CPUs");
+    cpuLog.print("                                                               s       %%  utilized");
+    cpuLog.print("   Process");
+    cpuLog.print("     Total                        %30.4f  %6.2f  %8.1f", process_cpu_time, 100.0, process_cpu_time / os::elapsedTime());
+    cpuLog.print("     VM Thread                    %30.4f  %6.2f  %8.1f", vm_thread_cpu_time, percent_of(vm_thread_cpu_time, process_cpu_time), vm_thread_cpu_time / os::elapsedTime());
+    cpuLog.print("     Garbage Collection           %30.4f  %6.2f  %8.1f", gc_cpu_time, percent_of(gc_cpu_time, process_cpu_time), gc_cpu_time / os::elapsedTime());
+    cpuLog.print("       GC Threads                 %30.4f  %6.2f  %8.1f", gc_threads_cpu_time, percent_of(gc_threads_cpu_time, process_cpu_time), gc_threads_cpu_time / os::elapsedTime());
+    cpuLog.print("       VM Thread                  %30.4f  %6.2f  %8.1f", gc_vm_thread_cpu_time, percent_of(gc_vm_thread_cpu_time, process_cpu_time), gc_vm_thread_cpu_time / os::elapsedTime());
+
+    if (UseStringDeduplication) {
+      double string_dedup_cpu_time = (double) gc_stats.stringdedup / NANOSECS_PER_SEC;
+      cpuLog.print("       String Deduplication       %30.4f  %6.2f  %8.1f", string_dedup_cpu_time, percent_of(string_dedup_cpu_time, process_cpu_time), string_dedup_cpu_time / os::elapsedTime());
     }
-    out.print("GC CPU usage: %.2f%% (Process: %.4fs GC: %.4fs)", usage, process_cpu_time, gc_cpu_time);
+    cpuLog.print("=====================================================================================");
   }
 }
 
 void CollectedHeap::before_exit() {
   print_tracing_info();
 
-  // Log GC CPU usage.
-  log_gc_cpu_time();
+  log_cpu_time();
```

Comment:
As mentioned offline, I would prefer if this was moved away from `CollectedHeap` since it is now not only GC/Heap related times that are tracked. 

We could add a `Universe::before_exit()` which firsts does the CPU logging before calling `heap()->before_exit()`. As long as we do the logging before, would there be any problem with this?